File: json/schema/libs/ts/src/json.ts
export type GtjAny = GtjNull | GtjBoolean | GtjNumber | GtjString | GtjArray | GtjObject | GtjUnion | GtjLiteral | GtjTuple;

export interface GtjBase {
  name?: string | undefined;
  doc?: string | undefined;
}

export interface GtjNull extends GtjBase {
  kind: "null";
}

export interface GtjBoolean extends GtjBase {
  kind: "boolean";
}

export interface GtjNumber extends GtjBase {
  kind: "number";
}

export interface GtjString extends GtjBase {
  kind: "string";
}

export interface GtjArray extends GtjBase {
  kind: "array";
  descriptor: GtjAny;
}

export interface GtjObject extends GtjBase {
  kind: "object";
  properties: Array<GtjProperty>;
}

export interface GtjProperty {
  kind: "property";
  name: string;
  doc?: string | undefined;
  descriptor: GtjAny;
  required: boolean;
}

export interface GtjUnion extends GtjBase {
  kind: "union";
  descriptors: Array<GtjAny>;
}

export interface GtjTuple extends GtjBase {
  kind: "tuple";
  descriptors: Array<GtjAny>;
}

export interface GtjLiteral extends GtjBase {
  kind: "literal";
  value: string | number | boolean | null;
}

export type GtjLiteralKind = "string" | "number" | "boolean" | "null";

File: json/tree/libs/py/genotype_json_tree/json.py
from typing import Union, TypeAlias, Optional, Literal, List
from genotype import Model


class GtjBase(Model):
    name: Optional[str] = None
    doc: Optional[str] = None


class GtjNull(GtjBase, Model):
    kind: Literal["null"]


class GtjBoolean(GtjBase, Model):
    kind: Literal["boolean"]


class GtjNumber(GtjBase, Model):
    kind: Literal["number"]


class GtjString(GtjBase, Model):
    kind: Literal["string"]


class GtjLiteral(GtjBase, Model):
    kind: Literal["literal"]
    value: Union[str, float, bool, None]


GtjLiteralKind: TypeAlias = Union[Literal["string"], Literal["number"], Literal["boolean"], Literal["null"]]


GtjAny: TypeAlias = Union[GtjNull, GtjBoolean, GtjNumber, GtjString, "GtjArray", "GtjObject", "GtjUnion", GtjLiteral, "GtjTuple"]


class GtjArray(GtjBase, Model):
    kind: Literal["array"]
    descriptor: GtjAny


class GtjObject(GtjBase, Model):
    kind: Literal["object"]
    properties: List["GtjProperty"]


class GtjProperty(Model):
    kind: Literal["property"]
    name: str
    doc: Optional[str] = None
    descriptor: GtjAny
    required: bool


class GtjUnion(GtjBase, Model):
    kind: Literal["union"]
    descriptors: List[GtjAny]


class GtjTuple(GtjBase, Model):
    kind: Literal["tuple"]
    descriptors: List[GtjAny]
